testsuite/smoke-test/src/network.rs

```diff
@@ -199,6 +261,40 @@ fn check_endpoint(
     result.is_ok()
 }
 
+fn network_address(node_config: &NodeConfig, network_id: &NetworkId) -> NetworkAddress {
+    let network = network(node_config, network_id);
+
+    let port = network
+        .listen_address
+        .as_slice()
+        .iter()
+        .find_map(|proto| {
+            if let Protocol::Tcp(port) = proto {
+                Some(port)
+            } else {
+                None
+            }
+        })
+        .unwrap();
+    let key = network.identity_key().public_key();
+    NetworkAddress::from_str(&format!(
+        "/ip4/127.0.0.1/tcp/{}/ln-noise-ik/{}/ln-handshake/0",
+        port, key
+    ))
+    .unwrap()
+}
+
+fn network<'a>(node_config: &'a NodeConfig, network_id: &NetworkId) -> &'a NetworkConfig {
+    match network_id {
+        NetworkId::Validator => node_config.validator_network.as_ref().unwrap(),
+        _ => node_config
+            .full_node_networks
+            .iter()
+            .find(|network| network.network_id == NetworkId::Public)
+            .unwrap(),
```

Comment:
That's kinda weird, considering the `NetworkId` isn't always public.  Maybe `== network_id`

Reply:
Mind elaborating what you mean?